Line/iOS/LINELoginDemo/LINELoginDemo/LineAuthManager.m

```diff
@@ -0,0 +1,230 @@
+        if (error)
+        {
+            [self returnSignInResult:nil error:error];
+        }
+    }
+}
+
+#pragma mark - Step 2: Communication with own validation server
+- (void)requestFirebaseAuthTokenWithLINEAccessToken:(NSString *)lineAccessToken {
+    // Stop current Firebase token fetcher if there's any
+    if (self.firebaseTokenFetcher && self.firebaseTokenFetcher.isFetching) {
+        [self.firebaseTokenFetcher stopFetching];
+    }
+    
+    // STEP 2: Exchange LINE access token for Firebase Custom Auth token
+    NSString *urlString = [NSString stringWithFormat:@"%@/verifyToken", YOUR_VALIDATION_SERVER_DOMAIN];
+    NSURL *url = [NSURL URLWithString:urlString];
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:@"application/json" forHTTPHeaderField:@"content-type"];
+    
+    NSDictionary *token = @{@"token" : lineAccessToken};
+    NSError *error;
+    NSData *requestBody = [NSJSONSerialization dataWithJSONObject:token
+                                                          options:kNilOptions error:&error];
+    if (error) {
+        [self returnSignInResult:nil error:error];
+    }
+    [request setHTTPBody:requestBody];
+    
+    self.firebaseTokenFetcher = [GTMHTTPFetcher fetcherWithRequest:request];
+    self.firebaseTokenFetcher.allowedInsecureSchemes = @[@"http"]; // Allow fetching http so that you can run your server sample code locally without https
+    __weak typeof(self) wSelf = self;
+    
+    [self.firebaseTokenFetcher beginFetchWithCompletionHandler:^(NSData *data, NSError *error) {
+        if (error) {
+            [wSelf returnSignInResult:nil error:error];
+        } else {
+            NSError *jsonSerializationError;
+            NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&jsonSerializationError];
+            
+            if (error) {
+                [wSelf returnSignInResult:nil error:jsonSerializationError];
+            } else {
+                NSString *firebaseToken = response[@"firebase_token"];
+                [wSelf authenticateWithFirebaseToken:firebaseToken];
+            }
+        }
+    }];
+}
+
+#pragma mark - Step 3: Communication with Firebase Auth SDK
+- (void)authenticateWithFirebaseToken:(NSString *)firebaseToken {
+    // STEP 3: Login to Firebase using Firebase Custom Auth token
+    [[FIRAuth auth] signInWithCustomToken:firebaseToken completion:^(FIRUser * _Nullable user, NSError * _Nullable error) {
+        if (error) {
+            [self returnSignInResult:nil error:error];
+            return;
+        }
+        
+        // Try to update
+        [self updateUserProfile:user];
+    }];
+}
+
+#pragma mark - Step 4: (Optional) Update user profile with LINE Profile
+- (void)updateUserProfile:(FIRUser *)user {
+    BOOL isProfileNeededUpdate = ([user.displayName length] == 0) || ([[user.photoURL absoluteString] length] == 0);
```

Comment:
Fixed in this [commit](https://github.com/firebase/custom-auth-samples/pull/1/commits/cfb5c9d622d615a16b4c27754be32543065571cc) :)